dvc/repo/artifacts.py

```diff
@@ -0,0 +1,91 @@
+import os
+from typing import TYPE_CHECKING
+
+from dvc.annotations import Artifact
+from dvc.dvcfile import FileMixin, ProjectFile
+
+if TYPE_CHECKING:
+    from dvc.repo import Repo
+
+
+class ArtifactsFile(FileMixin):
+    from dvc.schema import ARTIFACTS_SCHEMA as SCHEMA
+
+
+class Artifacts:
+    repo: "Repo"
+
+    def __init__(self, repo) -> None:
+        self.repo = repo
+
+    def _read(self):
+        # merge artifacts from all dvc.yaml files found
+        artifacts = {}
+        for dvcfile, dvcfile_artifacts in self.repo.index._artifacts.items():
+            # read the artifacts.yaml file if needed
+            if isinstance(dvcfile_artifacts, str):
+                dvcfile_artifacts = ArtifactsFile(
+                    self.repo,
+                    os.path.join(os.path.dirname(dvcfile), dvcfile_artifacts),
+                    verify=False,
+                ).load()
+            for name, value in dvcfile_artifacts.items():
+                if name in artifacts:
+                    # maybe better to issue a warning here and take the first one?
+                    # QUESTION: how to deal with the same artifact name in different dvc.yaml files?
```

Comment:
https://github.com/iterative/gto/issues/198#issuecomment-1480901662
cc @dberenbaum @shcheklein

Reply:
I think it should raise a duplicated artifact error.